btax/get_taxcalc_rates.py

```diff
@@ -0,0 +1,213 @@
+    --------------------------------------------------------------------
+    INPUTS:
+    baseline        = boolean, =True if baseline tax policy, =False if reform
+    start_year      = integer, first year of budget window
+    reform          = dictionary, reform parameters
+
+    OTHER FUNCTIONS AND FILES CALLED BY THIS FUNCTION: None
+
+    OBJECTS CREATED WITHIN FUNCTION:
+    micro_data_dict = dictionary, contains pandas dataframe for each year
+                      of budget window.  Dataframe contain mtrs, etrs, income variables, age
+                      from tax-calculator and PUF-CPS match
+
+    OUTPUT:
+        individual_rates: a dictionary of individual (IIT+payroll) marginal tax rates
+
+    RETURNS: individual_rates
+    --------------------------------------------------------------------
+    '''
+
+    calc1 = get_calculator(baseline=baseline, calculator_start_year=start_year,
+                           reform=reform)
+
+    # running all the functions and calculates taxes
+    calc1.calc_all()
+
+    ##get mtrs
+    # agi
+    #[mtr_fica_agi, mtr_iit_agi, mtr_combined_agi] = calc1.mtr('e00100')
+    # non-corprorate business income
+    # sch c
+    [mtr_fica_schC, mtr_iit_schC, mtr_combined_schC] = calc1.mtr('e00900p')
+    # sch e
+    [mtr_fica_schE, mtr_iit_schE, mtr_combined_schE] = calc1.mtr('e02000')
+    business_income_mtr = (sum([ abs(getattr(calc1.records, 'e00900')) * mtr_combined_schC
+                           + abs(getattr(calc1.records, 'e02000')) * mtr_combined_schE])/
+                            (sum(map(abs,getattr(calc1.records, 'e02000'))) +
+                            sum(map(abs,getattr(calc1.records, 'e02000')))))
+    # dividends
+    [mtr_fica_div, mtr_iit_div, mtr_combined_div] = calc1.mtr('e00650')
+    # interest income
+    # taxable
+    [mtr_fica_int, mtr_iit_int, mtr_combined_int] = calc1.mtr('e00300')
+    # non-taxable
+    [mtr_fica_int_te, mtr_iit_int_te, mtr_combined_int_te] = calc1.mtr('e00400')
+    interest_income_mtr = (sum([ abs(getattr(calc1.records, 'e00300')) * mtr_combined_int
+                           + abs(getattr(calc1.records, 'e00400')) * mtr_combined_int_te])/
+                            (sum(map(abs,getattr(calc1.records, 'e00300'))) +
+                            sum(map(abs,getattr(calc1.records, 'e00400')))))
+    # short term capital gains
+    [mtr_fica_scg, mtr_iit_scg, mtr_combined_scg] = calc1.mtr('p22250')
+    # long term capital gains
+    [mtr_fica_lcg, mtr_iit_lcg, mtr_combined_lcg] = calc1.mtr('p23250')
+    # pension distributions
+    # does PUF have e01500?  Do we want IRA distributions here?
+    # Weird - I see e01500 in PUF, but error when try to call it
+    [mtr_fica_pension, mtr_iit_pension, mtr_combined_pension] = calc1.mtr('e01700')
+    # mortgage interest and property tax deductions
+    # do we also want mtg ins premiums here?
+    # mtg interest
+    [mtr_fica_mtg, mtr_iit_mtg, mtr_combined_mtg] = calc1.mtr('e19200')
```

Comment:
Thanks for finding that @Amy-Xu.  I recall something like this, but couldn't find the lines of code to point out.  Does this limitation of the valid strings have to do with some TaxCalc logic or for another reason (e.g. to save computational time it does not compute MTRs for every line item on the 1040)?  

Relatedly, there are some variables such as tax exempt interest that is (obviously) zero-taxed under current law.  But one could imagine a reform where the tax-exempt status of this income source were to change.  How would one handle that?  In particular, would the Tax Calc output then put all interest income in  e00300?  Or would the formerly tax exempt still show up under e00400, but now face the non-zero tax?  I'm just trying to think about how I might compute a MTR on interest income as a whole (taking into account that some might be taxable and some not, with the taxable status changing between the baseline and reform).

cc @MattHJensen